src/components/custom-field-input-text/custom-field-input-text.jsx

```diff
@@ -0,0 +1,67 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import Icon from '../icon/icon.jsx';
+import styles from './custom-field-input-text.css';
+
+function getRootClassName(className, error, disabled) {
+  const resolvedClassName = className || styles['custom-field-input-text'];
+
+  if (disabled) {
+    return `${resolvedClassName} ${styles.disabled}`;
```

Comment:
RE: Do we want to fully utilize CSS Composition through CSS Modules or concatenate our default styles?

I think we need to look at a few different examples:

SOON TO COME

Reply:
Possibly, we might want design to give us their opinion on how a CSS API should interact with the defaults.